gooddata-java/src/test/java/com/gooddata/sdk/service/hierarchicalconfig/HierarchicalConfigServiceIT.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (C) 2004-2021, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.sdk.service.hierarchicalconfig;
+
+import com.gooddata.sdk.model.hierarchicalconfig.SettingItem;
+import com.gooddata.sdk.model.hierarchicalconfig.Settings;
+import com.gooddata.sdk.model.project.Project;
+import com.gooddata.sdk.service.AbstractGoodDataIT;
+import org.springframework.web.util.UriTemplate;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import static com.gooddata.sdk.common.util.ResourceUtils.readObjectFromResource;
+import static com.gooddata.sdk.common.util.ResourceUtils.readStringFromResource;
+import static net.jadler.Jadler.onRequest;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.hamcrest.Matchers.is;
+
+public class HierarchicalConfigServiceIT extends AbstractGoodDataIT {
+
+    private HierarchicalConfigService service;
+    private Project project;
+
+    private static final UriTemplate PROJECT_SETTINGS_TEMPLATE = Settings.PROJECT_SETTINGS_TEMPLATE;
+    private static final UriTemplate PROJECT_SETTING_TEMPLATE = SettingItem.PROJECT_SETTING_TEMPLATE;
+    private static final String PROJECT_ID = "PROJECT_ID";
+    private static final String PROJECT_SETTINGS_URI_STRING = PROJECT_SETTINGS_TEMPLATE.expand(PROJECT_ID).toString();
+    private static final String SETTING_ITEM_NAME = "myCoolFeature";
+    private static final String PROJECT_SETTING_URI_STRING = PROJECT_SETTING_TEMPLATE
+            .expand(PROJECT_ID, SETTING_ITEM_NAME).toString();
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        service = gd.getHierarchicalConfigService();
+        project = readObjectFromResource("/project/project.json", Project.class);
+    }
+
+    @Test
+    public void projectSettingsShouldReturnProjectSettings() {
+        onRequest()
+                .havingMethodEqualTo("GET")
+                .havingPathEqualTo(PROJECT_SETTINGS_URI_STRING)
+                .respond()
+                .withBody(readStringFromResource("/hierarchicalconfig/settings.json"))
+                .withStatus(200);
+
+        final Settings settings = service.projectSettings(project);
+
+        assertThat(settings, containsInAnyOrder(
+                new SettingItem("myCoolFeature", "true", "project"),
+                new SettingItem("mySuperCoolFeature", "true", "project"),
+                new SettingItem("mySuperSecretFeature", "false", "project"),
+                new SettingItem("myNotBooleanSetting", "Disabled", "catalog")));
+    }
+
+    @Test
+    public void createProjectSettingShouldCreateAndReturnSameSetting() {
```

Comment:
Please use latest naming in method names

Reply:
Changed